### ISSUE TITLE
[Bug]: Provider producing inconsistent final plan - aws_s3_bucket and default_tags at provider

### ISSUE THREAD

**Issue by Mralbert93** [hashicorp/terraform-provider-aws#30801]
### Terraform Core Version

1.4.2

### AWS Provider Version

4.63.0

### Affected Resource(s)

aws_s3_bucket

### Expected Behavior

S3 bucket should be able to be created when default_tags are configured at provider level. 

### Actual Behavior

Run errors during apply phase. 

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for aws_s3_bucket.example to include new values
│ learned so far during apply, provider "registry.terraform.io/hashicorp/aws"
│ produced an invalid new value for .tags_all: new element "created" has
│ appeared.
│
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
Operation failed: failed running terraform apply (exit 1)
```

### Terraform Configuration Files

```
provider "aws" {
  region = "us-east-1"
  default_tags {
    tags = {
      created = timestamp()
    }
  }
}

locals {
  tags = {
    environment = "test"
    type        = "new"
  }
}

resource "aws_s3_bucket" "example" {
  bucket = "my-bucket-for-testing-broken-things"
  tags = local.tags
}
```

### Steps to Reproduce

1. `terraform init `
2. `terraform apply-auto-approve`

### References

https://github.com/hashicorp/terraform-provider-aws/issues/29842

### Would you like to implement a fix?

None

**Comment by bion** [comment]
Seeing a similar error:

```
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for module.ecs_fargate_service.module.ecs-alb[0].module.lb_logs_s3[0].aws_s3_bucket.logs to
│ include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid
│ new value for .tags_all: new element "Name" has appeared.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for module.ecs_fargate_service.module.ecs-alb[0].module.lb_logs_s3[0].aws_s3_bucket.logs to
│ include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid
│ new value for .tags_all: new element "Type" has appeared.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
```

Our code is open source and visible here: https://github.com/civiform/cloud-deploy-infra/blob/main/cloud/aws/templates/aws_oidc/filestorage.tf#L95

**Comment by johnsonaj** [comment]
we have done work to allow resource `tags` to be `computed` values. This update will be released in `v5.0.0` of the provider. 

It is still not possible to have `computed` values on the provider level `default_tags` block. This is due to limitations on how the provider is configured, and how these `default_tags` are applied to resources downstream.

The following configuration will be possible in `v5.0.0`
```terraform
provider "aws" {
  region = "us-west-2"
  default_tags {
    tags = {
        key1 = "value1"
    }
}

resource "aws_s3_bucket" "example" {
  bucket = "example-bucket"
  tags = {
    created_at = timestamp()
  }
}
```

However, the follow will still not compute correctly because `default_tags` must be known values

```terraform
provider "aws" {
  region = "us-west-2"
  default_tags {
    tags = {
        created_at = timestamp()
    }
  }
}

resource "aws_s3_bucket" "example" {
  bucket = "example-bucket"
  tags     = {
    additional_tag = "value"
  }
}
```

**Comment by jar-b** [comment]
Related update - the `plantimestamp` function (available with Terraform 1.5+) **can** be used with `default_tags` as the value will be known at apply time. [This comment](https://github.com/hashicorp/terraform-provider-aws/issues/35725#issuecomment-1946285906) includes a minimal reproduction demonstrating the plan/apply output with this approach.
